### ISSUE TITLE
Error occurred when setting M300 camera photo file format

### ISSUE THREAD

**Issue by gyungdal** [dji-sdk/Mobile-SDK-Android#735]
## Environment

|Name|Desc|
|-|-|
|SDK Version|4.14 trial1|
|Drone|[Matrice 300 RTK](https://www.dji.com/kr/matrice-300)|
|camera|[H20T](https://www.dji.com/kr/zenmuse-h20-series?) and [XT2](https://www.dji.com/zenmuse-xt2)|
|Firmware version|01.00.0214|

## Description
Hi, We are modifying the code used in the old version to use the new version of the SDK.
During code modification, I found that an error occurred when trying to set the photo file format.

```kotlin
camera?.getPhotoFileFormat(object : CommonCallbacks.CompletionCallbackWith<SettingsDefinitions.PhotoFileFormat>{
    override fun onSuccess(p0: SettingsDefinitions.PhotoFileFormat?) {
        //TODO("Not yet implemented")
        Log.i("Camera", "Support Camera Format : ${p0?.name} - ${p0?.ordinal}")
        camera.setPhotoFileFormat(SettingsDefinitions.PhotoFileFormat.JPEG, CommonCallbacks.CompletionCallback<DJIError> {
            ToastUtils.setFuncResultToToast(this.javaClass.simpleName, it)
            when (it) {
                null -> handler.postDelayed(Init8_CameraMediaFileCallback(handler), defaultSuccessDelay)
                else -> handler.postDelayed(this@Init7_CameraImageFormat, defaultFailedDelay)
            }
        })
    }

    override fun onFailure(p0: DJIError?) {
        Log.e("Camera", "Error : ${p0?.errorCode} - ${p0?.description}")
    }

})
```
So I tried to check the supported formats using the code above, but I got an error even at this code.
Is there any way to fix the this problem in the current SDK version?

### Error photo
![image](https://user-images.githubusercontent.com/3604032/103868588-6ae85300-510c-11eb-8c5c-9c644d9d1050.png)

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/42809'>Zendesk ticket #42809</a>:<br><br>尊敬的开发者，
您好，感谢您联系DJI 大疆创新。

M300+H20T的话，你进入到lens里再调用setPhotoFileFormat试试。

希望我们的解决方案能够帮到您，感谢您的邮件，祝您生活愉快！
Best Regards,
DJI 大疆创新SDK技术支持

**Comment by gyungdal** [comment]
Thank you.

I modified it as follows by referring to the answer.
but... application shows an server error as shown below.
How can I fix that?

```kotlin
        val lens = camera?.lenses?.last()
        lens?.getPhotoFileFormat(object : CommonCallbacks.CompletionCallbackWith<SettingsDefinitions.PhotoFileFormat>{
            override fun onSuccess(p0: SettingsDefinitions.PhotoFileFormat?) {
                //TODO("Not yet implemented")
                Log.i("Camera", "Support Camera Format : ${p0?.name} - ${p0?.ordinal}")
                lens.setPhotoFileFormat(SettingsDefinitions.PhotoFileFormat.RADIOMETRIC_JPEG, CommonCallbacks.CompletionCallback<DJIError> {
                    ToastUtils.setFuncResultToToast(this.javaClass.simpleName, it)
                    when (it) {
                        null -> handler.postDelayed(Init8_CameraMediaFileCallback(handler), defaultSuccessDelay)
                        else -> handler.postDelayed(this@Init7_CameraImageFormat, defaultFailedDelay)
                    }
                })
            }

            override fun onFailure(p0: DJIError?) {
                Log.e("Camera", "Error : ${p0?.errorCode} - ${p0?.description}")
            }

        })
        if (camera == null) {
            ToastUtils.setResultToToast(this.javaClass.simpleName + ": camera is null goto wait connect")
            handler.postDelayed(InitFailed(handler), defaultFailedDelay)
        }
```

![image](https://user-images.githubusercontent.com/3604032/103993836-8a997d00-51d9-11eb-913d-b1c2ff624c27.png)

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/42809'>Zendesk ticket #42809</a>:<br><br>Dear Client
Thank you for contacting DJI.

I can see you have gotton your file foramt. Can you tell me which method throws this error? Please be mind that you have to set the camera settings not to often. E.g. You have modify a camera setting and you want to capture a photo. You have to wait for like 2-3 seconds.

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support

**Comment by gyungdal** [comment]
A bug occurs when call the setPhotoFileFormat method.

```kotlin
class Init7_CameraImageFormat(handler: DroneMainHandler) : Initializer(handler) {
    override fun run() {
        Log.i("Init", "${this.javaClass.name}")
        val camera = DroneApplication.camera
        val lens = camera?.lenses?.last()
        lens?.getPhotoFileFormat(object : CommonCallbacks.CompletionCallbackWith<SettingsDefinitions.PhotoFileFormat>{
            override fun onSuccess(p0: SettingsDefinitions.PhotoFileFormat?) {
                //TODO("Not yet implemented")
                Log.i("Camera", "Support Camera Format : ${p0?.name} - ${p0?.ordinal}")
                lens.setPhotoFileFormat(SettingsDefinitions.PhotoFileFormat.RADIOMETRIC_JPEG, CommonCallbacks.CompletionCallback<DJIError> {
                    ToastUtils.setFuncResultToToast(this.javaClass.simpleName, it)
                    Log.i("Camera", "Set Photo File Format")
                    when (it) {
                        null -> handler.postDelayed(Init8_CameraMediaFileCallback(handler), defaultSuccessDelay)
                        else -> {
                            Log.i("Camera", "set photo file format error")
                            Log.i("camera", "${it.errorCode} : ${it.description}")
                            handler.postDelayed(InitFailed(handler), defaultFailedDelay)
                        }
                    }
                })
            }
```
![image](https://user-images.githubusercontent.com/3604032/104262843-c4190380-54cb-11eb-9022-699344d2dda5.png)


Camera setup proceeded as follows.
1. Camera mode setting
2. Image file format setting (a bug occurs here)

Have something that needs to be set further before?

**Comment by gyungdal** [comment]
The same problem occurs even if create a waiting time of 2-3 seconds using the Thread.sleep or handler.postDelayed method.

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/42809'>Zendesk ticket #42809</a>:<br><br>Dear Client
Thank you for contacting DJI.

I have tested my side and I found error too. But my error message is different to you, mine shows exection time out. I already report this bug to the dev team.

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support
